src/network/postgres/postgres_protocol_interpreter.cpp

```diff
@@ -27,11 +28,12 @@ Transition PostgresProtocolInterpreter::Process(std::shared_ptr<ReadBuffer> in,
     curr_input_packet_.Clear();
     return ProcessStartup(in, out);
   }
-  std::shared_ptr<PostgresNetworkCommand> command = command_factory_->PostgresPacketToCommand(&curr_input_packet_);
+  std::shared_ptr<PostgresNetworkCommand> command =
+      std::static_pointer_cast<PostgresNetworkCommand>(command_factory_->PacketToCommand(&curr_input_packet_));
   PostgresPacketWriter writer(out);
   if (command->FlushOnComplete()) out->ForceFlush();
-  Transition ret = command->Exec(common::ManagedPointer(this), common::ManagedPointer(&writer), t_cop,
-                                 common::ManagedPointer(context), callback);
+  Transition ret = command->Exec(common::ManagedPointer<ProtocolInterpreter>(this),
+                                 common::ManagedPointer<PostgresPacketWriter>(&writer), t_cop, context, callback);
```

Comment:
@tanujnay112 @tli2 Why does the `ForceFlush` occur before the command is executed? Shouldn't it happen after?